flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -53,12 +54,13 @@ public static CheckpointBarrierHandler createCheckpointBarrierHandler(
 					+ " must be positive or -1 (infinite)");
 			}
 
+			int pageSize = NetworkEnvironmentConfiguration.getPageSize(taskManagerConfig);
```

Comment:
Are we going to have page size and `NETWORK_CREDIT_MODEL` for all shuffle services? Or is there any other better way how to abstract BarrierBuffer creation, like asking NetworkEnvironment/ShuffleService about needed parameters and pass them into Environment/StreamTasks?

Reply:
Thanks for this good catch. 

The options of page size and `NETWORK_CREDIT_MODEL` might be not general for all shuffle services. `BarrierHandler` should not concern of the specific shuffle service implementation and see these parameters in theory.

Actually the `NETWORK_CREDIT_MODEL` option is deprecated. If we could remove this option completed and the page size is also not needed because `CachedBufferBlocker` could get the page size from `BufferOrEvent#getSize` after introducing the `InputGateWithMetrics` in another PR. 

There was a blocker for removing these deprecated codes before, but I think this blocker is already confirmed now. Then I would talk with Piotr about when to remove them (would be better if done in release 1.9). I think this is a right way to go to avoid extra efforts of abstracting `BarrierBuffer` creation. WDYT?